client/X11/xf_client.c

```diff
@@ -1076,8 +1076,9 @@ static const button_map xf_button_flags[NUM_BUTTONS_MAPPED] = {
 	{ Button2, PTR_FLAGS_BUTTON3 },
 	{ Button3, PTR_FLAGS_BUTTON2 },
 	{ Button4, PTR_FLAGS_WHEEL | 0x78 },
-	{ Button5, PTR_FLAGS_WHEEL | PTR_FLAGS_WHEEL_NEGATIVE | 0x78 },
-	{ 6, PTR_FLAGS_HWHEEL | PTR_FLAGS_WHEEL_NEGATIVE | 0x78 },
+	{ Button5, PTR_FLAGS_WHEEL | PTR_FLAGS_WHEEL | 0x78 },
+	/* Negative value is 9bit twos complement */
+	{ 6, PTR_FLAGS_HWHEEL | PTR_FLAGS_WHEEL | (0x100 - 0x78) },
```

Comment:
This is now also wrong: The documentation says:
```
The only valid flags in a horizontal wheel rotation event are PTRFLAGS_WHEEL_NEGATIVE and the
WheelRotationMask
```
It also mentions:
```
If both PTRFLAGS_WHEEL and PTRFLAGS_HWHEEL are specified, then PTRFLAGS_WHEEL takes
precedence.
```

So, you need to remove the `PTR_FLAGS_WHEEL` here.